cluster/addons.go

```diff
@@ -39,14 +45,113 @@ func (c *Cluster) deployUserAddOns(ctx context.Context) error {
+		return c.deployAddonsInclude(ctx)
+	}
+	log.Infof(ctx, "[addons] User addon deployed successfully..")
+	if err := c.doAddonDeploy(ctx, c.Addons, UserAddonResourceName); err != nil {
+		return err
+	}
+
+	return c.deployAddonsInclude(ctx)
+}
+
+// deployAddonsInclude takes any passed addons from the addons key and appends yaml contents from urls/paths
+// then deploys all addons at once. Not doing this in this manner will fail to deploy all of the plugins.
+// Only the addons from the addons: key would be deployed.
+func (c *Cluster) deployAddonsInclude(ctx context.Context) error {
+	var userAddons = ""
+	log.Infof(ctx, "[addons] Checking for included user addons")
+
+	yamlDelimiter := regexp.MustCompile("(\n---$|\n---\n$)")
+	// If there are contents in the array continue
+	if len(c.AddonsInclude) != 0 {
+
+		for _, addon := range c.AddonsInclude {
+			// Check if the string is a url
+			// If it is read the response body and pass it as a string to doAddonDeploy
+			if strings.HasPrefix(addon, "http") {
+				addonYAML, err := c.getAddonFromUrl(addon)
+
+				if err != nil {
+					return err
+				}
+
+				log.Infof(ctx, "[addons] Adding addon from url %s", addon)
+
+				// Debug statements to troubleshoot addons not deploying correctly
+				logrus.Debugf("URL Yaml: %s", addonYAML)
+
+				// Some manifests contain --- at the end of the file, if it does we do not need to append it
+				if yamlDelimiter.MatchString(addonYAML) {
+					log.Infof(ctx, "[addons] Found suffix: %s", addonYAML)
+					userAddons += addonYAML
+				} else {
+					userAddons += fmt.Sprintf("%s\n---\n", addonYAML)
+				}
+
+			} else if c.isFilePath(addon) {
+				// If the entry is a filepath that exists read it and convert []byte to string
+				yamlContents, err := ioutil.ReadFile(addon)
+
+				if err != nil {
+					return err
+				}
+
+				log.Infof(ctx, "[addons] Adding addon from %s", addon)
+
+				// Debug statements to troubleshoot addons not deploying correctly
+				logrus.Debugf("FilePath Yaml: %s", yamlContents)
+
+				// Some manifests contain --- at the end of the file, if it does we do not need to append it
+				if yamlDelimiter.MatchString(string(yamlContents)){
+					userAddons += string(yamlContents)
+				} else {
+					userAddons += fmt.Sprintf("%s\n---\n", string(yamlContents))
+				}
+			} else {
+				log.Warnf(ctx, "[addons] Unable to determine if %s is a filepath or url, skipping", addon)
+			}
+
+		}
+
+	} else {
+		log.Infof(ctx, "[addons] No included addon paths or urls..")
 		return nil
 	}
 
-	if err := c.doAddonDeploy(ctx, c.Addons, UserAddonResourceName); err != nil {
+	log.Infof(ctx, "[addons] Deploying %s", UserAddonsIncludeResourceName)
+	logrus.Debugf("[addons] Compiled addons yaml: %s", userAddons)
+	if err := c.doAddonDeploy(ctx, userAddons, UserAddonsIncludeResourceName); err != nil {
 		return err
 	}
-	log.Infof(ctx, "[addons] User addon deployed successfully..")
+
 	return nil
+}
+
+func (c *Cluster) isFilePath(addonPath string) bool {
```

Comment:
i am not sure if this function should be a method struct since its not related to the cluster object at all